File: Hackerrank/30_days_of_code_cpp/day_21.cpp
#include <iostream>
#include <vector>

using namespace std;

template <typename T>

void printArray(vector<T> A) {
    for (int i = 0; i < A.size(); i++) {
        cout << A[i] << endl;
    }
}

int main() {
    vector<int> vector_int{1, 2, 3};
    vector<string> vector_str{"Hello", "World"};
    printArray<int>(vector_int);
    printArray<string>(vector_str);
    return 0;
}

File: Hackerrank/30_days_of_code_csharp/day_21.cs
using System;

public class Solution {

    public static void printArray<T>(T[] A){
       for (int i = 0; i < A.Length; i++) {
           Console.WriteLine(A[i]);
       }
    }

    static void Main(string[] args) {
        int[] vInt = new int[] {1, 2, 3};
        string[] vString = new string[] {"Hello", "World"};

        printArray<int>(vInt);
        printArray<string>(vString);
    }
}
